Connectors/Rhino/Speckle.Connectors.RhinoShared/Bindings/RhinoMapperBinding.cs

```diff
@@ -134,18 +189,118 @@ public CategoryMapping[] GetCurrentMappings()
+    }
+
+    // Trigger single update
+    _idleManager.SubscribeToIdle(nameof(NotifyMappingsChanged), NotifyMappingsChanged);
+  }
 
   /// <summary>
-  /// Converts a string object ID to a RhinoObject.
+  /// Removes category assignments from specific layer(s).
   /// </summary>
-  /// <returns>RhinoObject if found and valid, null otherwise</returns>
-  /// <remarks>Reducing repetitive code.</remarks>
-  private static RhinoObject? GetRhinoObject(string objectIdString) =>
-    Guid.TryParse(objectIdString, out var objectId) ? RhinoDoc.ActiveDoc.Objects.FindId(objectId) : null;
+  public void ClearLayerCategoryAssignment(string[] layerIds)
+  {
+    foreach (var layerId in layerIds)
+    {
+      // NOTE: clear user string by setting to null. Layer has not DeleteUserString() method 🙄
+      var layer = GetLayer(layerId);
+      layer?.SetUserString(RevitMappingConstants.CATEGORY_USER_STRING_KEY, null);
+    }
+
+    // Trigger single update
+    _idleManager.SubscribeToIdle(nameof(NotifyMappingsChanged), NotifyMappingsChanged);
+  }
+
+  /// <summary>
+  /// Removes all layer category assignments in the doc.
+  /// </summary>
+  public void ClearAllLayerCategoryAssignments()
+  {
+    foreach (var layer in RhinoDoc.ActiveDoc.Layers)
+    {
+      if (!string.IsNullOrEmpty(layer.GetUserString(RevitMappingConstants.CATEGORY_USER_STRING_KEY)))
+      {
+        // NOTE: clear user string by setting to null. Layer has not DeleteUserString() method 🙄
+        layer.SetUserString(RevitMappingConstants.CATEGORY_USER_STRING_KEY, null);
+      }
+    }
+
+    // Trigger single update
+    _idleManager.SubscribeToIdle(nameof(NotifyMappingsChanged), NotifyMappingsChanged);
+  }
+
+  /// <summary>
+  /// Gets all current layer mappings to show in the UI table.
+  /// Layers with the same category mapping are grouped together.
+  /// </summary>
+  public LayerCategoryMapping[] GetCurrentLayerMappings()
+  {
+    var mappedLayers = RhinoDoc
+      .ActiveDoc.Layers.Where(layer =>
+        !string.IsNullOrEmpty(layer.GetUserString(RevitMappingConstants.CATEGORY_USER_STRING_KEY))
+      )
+      .GroupBy(layer => layer.GetUserString(RevitMappingConstants.CATEGORY_USER_STRING_KEY))
+      .Select(group => new LayerCategoryMapping(
+        group.Key,
+        RevitBuiltInCategoryStore.GetLabel(group.Key),
+        group.Select(layer => layer.Id.ToString()).ToArray(),
+        group.Select(layer => GetFullLayerPath(layer)).ToArray(),
+        group.Count()
+      ))
+      .ToArray();
+
+    return mappedLayers;
+  }
+
+  /// <summary>
+  /// Gets all objects that would effectively receive the specified layer mapping during send.
+  /// Takes into account hierarchical resolution - only returns objects that would actually
+  /// resolve to this specific category value through the layer hierarchy.
+  /// </summary>
+  public string[] GetEffectiveObjectsForLayerMapping(string[] layerIds, string categoryValue)
```

Comment:
can go into `RevitMappingResolver`

Reply:
[refactor: move GetEffectiveObjectsForLayerMapping to `RevitMappingR…](https://github.com/specklesystems/speckle-sharp-connectors/pull/1027/commits/4067af542d1679964270bfce2de96bf0486600e8)